### ISSUE TITLE
Not all characters supported in root element name

### ISSUE THREAD

**Issue by ptrckb** [INRIA/tralics#15]
The code for the `DocType` command (`MainClass::find_dtd()`, https://github.com/tralics/tralics/blob/307104b23be1fe5c620df256cf604e0e14c9d670/src/MainClass.cpp#L804) is restricting the characters authorized for the name of the root element (https://www.w3.org/TR/2008/REC-xml-20081126/#d0e804).
For example it does not recognize the '-' character.

**Comment by vbeffara** [comment]
Good catch. It's an easy fix on the regexp, but will require cautious testing.

The previous version of that code looked more handcrafted, like https://github.com/tralics/tralics/blob/refactor/src/Buffer.cpp#L34 ...
